Sources/TuistGraph/Models/PlatformFilter.swift

```diff
@@ -7,12 +7,15 @@ extension PlatformFilters: Comparable {
     public static func < (lhs: Set<Element>, rhs: Set<Element>) -> Bool {
         lhs.map(\.xcodeprojValue).sorted().joined() < rhs.map(\.xcodeprojValue).sorted().joined()
     }
+
+    public static let all = Set(PlatformFilter.allCases)
+    public static let invalid = Set<PlatformFilter>()
```

Comment:
no filters doesn't necessarily mean invalid filters - perhaps to simplify we can make `platformFilters()` method in graph traverser return a dedicated type

```swift
enum PlatformFiltersResult {
  case disjoint
  case filters(PlatformFilters)
}
```

And `dependencyReference` can prune accordingly

Reply:
My intent with this pass was to reduce the need for an extra type here,  having `.all == []` makes the implementation harder generally as well.  

Currently `[GraphEdge: PlatformFilters]` will return `.all` when there is not a configured set of platform filters.  My hope was to align the rest of our implementation with this assumption such that we can just use normal set operations to resolve what we need to avoid the extra complexity of that type.  

What cases are we seeing things unnecessarily pruned?  there may still be some spots from the previous iteration that are explicitly setting `[]` in `graph.dependencyPlatformFilters` that need to be changed to `.all`